editor/src/messages/portfolio/document/document_message_handler.rs

```diff
@@ -483,14 +496,14 @@ impl MessageHandler<DocumentMessage, DocumentInputs<'_>> for DocumentMessageHand
 					id: folder_id,
 					nodes: HashMap::new(),
 					parent,
-					insert_index: -1,
+					insert_index: calculated_insert_index.unwrap_or(-1),
 				});
 				responses.add(PortfolioMessage::PasteIntoFolder {
 					clipboard: Clipboard::Internal,
 					parent: LayerNodeIdentifier::new_unchecked(folder_id),
 					insert_index: -1,
 				});
-
+				let folder_id = NodeId(generate_uuid()); // TODO: Either this is a bug, or its reasoning should be explained in a comment
```

Comment:
This line looks like it was accidentally copied from above (line 490) and put here. But if it has a purpose, please explain it for me so we can discuss if it should be kept.

Reply:
Yes it's a mistake